test/unit/modules/transport.js

```diff
@@ -369,86 +383,234 @@ describe('transport', function () {
+				library = {
+					schema: {
+						validate: sinonSandbox.stub().callsArg(2)
+					},
+					logger: {
+						debug: sinonSandbox.spy()
+					}
+				};
+
+				modules = {
+					peers: {
+						remove: sinonSandbox.stub().returns(true)
+					}
+				};
+
+				restoreRewiredDeps = TransportModule.__set__({
+					library: library,
+					modules: modules
+				});
 
-			it('should call library.schema.validate with query');
+				query = {
+					transactions: [
+						{
+							id: '222675625422353767',
+							type: 0,
+							amount: '100',
+							fee: '10',
+							senderPublicKey: '2ca9a7143fc721fdc540fef893b27e8d648d2288efa61e56264edf01a2c23079',
+							recipientId: '12668885769632475474L',
+							timestamp: 28227090,
+							asset: {},
+							signature: '2821d93a742c4edf5fd960efad41a4def7bf0fd0f7c09869aed524f6f52bf9c97a617095e2c712bd28b4279078a29509b339ac55187854006591aa759784c205'
+						}
+					]
+				};
 
-			it('should call library.schema.validate with schema.transactions');
+				__private.receiveTransaction = sinonSandbox.stub().callsArg(3);
 
-			describe('when library.schema.validate fails', function (){
+				done();
+			});
 
-				it('should call callback with error = "Invalid transactions body"');
+			afterEach(function (done) {
+				restoreRewiredDeps();
+				done();
 			});
 
-			describe('when library.schema.validate succeeds', function (){
+			// TODO: It doesn't seem that library.schema.validate currently gets called by the __private.receiveTransaction logic.
+			// it('should call library.schema.validate with query and definitions.Transaction', function (done) {
+			// 	__private.receiveTransactions(query, peerStub, '', function (err) {
+			// 		expect(err).to.equal(null);
+			// 		expect(library.schema.validate.calledWith(query, defaultScope.swagger.definitions.Transaction)).to.be.true;
+			// 		done();
+			// 	});
+			// });
+			//
+			// describe('when library.schema.validate fails', function () {
+			//
+			// 	it('should call callback with error = "Invalid transactions body"', function (done) {
+			// 		var validateErr = new Error('Transaction did not match schema');
+			// 		validateErr.code = 'INVALID_FORMAT';
+			// 		library.schema.validate = sinonSandbox.stub().callsArgWith(2, [validateErr]);
+			//
+			// 		__private.receiveTransactions(query, peerStub, '', function (err) {
+			// 			expect(library.schema.validate.called).to.be.true;
+			// 			// TODO: Check that err is what we expect it to be.
+			// 			done();
+			// 		});
+			// 	});
+			// });
+
+			describe('when library.schema.validate succeeds', function () {
 
 				describe('for every transaction in transactions', function () {
 
 					describe('when transaction is undefined', function () {
 
-						it('should call callback with error = "Unable to process signature. Signature is undefined."');
+						it('should call callback with error = "Unable to process transaction. Transaction is undefined."', function (done) {
+							query.transactions[0] = undefined;
```

Comment:
How about moving it to newly created `beforeEach` block, so it takes a responsibility of what's happening inside its scope.

Reply:
Ok I'll put it in a `before` block. I can't really use `beforeEach` here because this stub should only affect this one test case.